probe_src/probe_frontend/python/ops.py

```diff
@@ -0,0 +1,361 @@
+        self.stx_mask = int(stx_mask)
+        self.stx_blksize = int(stx_blksize)
+        self.stx_attributes = int(stx_attributes)
+        self.stx_nlink = int(stx_nlink)
+        self.stx_uid = int(stx_uid)
+        self.stx_gid = int(stx_gid)
+        self.stx_mode = int(stx_mode)
+        self.stx_ino = int(stx_ino)
+        self.stx_size = int(stx_size)
+        self.stx_blocks = int(stx_blocks)
+        self.stx_attributes_mask = int(stx_attributes_mask)
+        self.stx_atime = statx_timestamp(**stx_atime)
+        self.stx_btime = statx_timestamp(**stx_btime)
+        self.stx_ctime = statx_timestamp(**stx_ctime)
+        self.stx_mtime = statx_timestamp(**stx_mtime)
+        self.stx_rdev_major = int(stx_rdev_major)
+        self.stx_rdev_minor = int(stx_rdev_minor)
+        self.stx_dev_major = int(stx_dev_major)
+        self.stx_dev_minor = int(stx_dev_minor)
+        self.stx_mnt_id = int(stx_mnt_id)
+        self.stx_dio_mem_align = int(stx_dio_mem_align)
+        self.stx_dio_offset_align = int(stx_dio_offset_align)
+
+@dataclass(init=False)
+class timeval:
+    tv_sec: int
+    tv_usec: int
+    def __init__(self, tv_sec: int, tv_usec: int):
+        self.tv_sec = int(tv_sec)
+        self.tv_usec = int(tv_usec)
+
+@dataclass(init=False)
+class rusage:
+    ru_utime: timeval
+    ru_stime: timeval
+    ru_maxrss: int
+    ru_ixrss: int
+    ru_idrss: int
+    ru_isrss: int
+    ru_minflt: int
+    ru_majflt: int
+    ru_nswap: int
+    ru_inblock: int
+    ru_oublock: int
+    ru_msgsnd: int
+    ru_msgrcv: int
+    ru_nsignals: int
+    ru_nvcsw: int
+    ru_nivcsw: int
+    def __init__(self, ru_utime: timeval, ru_stime: timeval, ru_maxrss: int, ru_ixrss: int, ru_idrss: int, ru_isrss: int, ru_minflt: int, ru_majflt: int, ru_nswap: int, ru_inblock: int, ru_oublock: int, ru_msgsnd: int, ru_msgrcv: int, ru_nsignals: int, ru_nvcsw: int, ru_nivcsw: int):
+        self.ru_utime = timeval(**ru_utime)
+        self.ru_stime = timeval(**ru_stime)
+        self.ru_maxrss = int(ru_maxrss)
+        self.ru_ixrss = int(ru_ixrss)
+        self.ru_idrss = int(ru_idrss)
+        self.ru_isrss = int(ru_isrss)
+        self.ru_minflt = int(ru_minflt)
+        self.ru_majflt = int(ru_majflt)
+        self.ru_nswap = int(ru_nswap)
+        self.ru_inblock = int(ru_inblock)
+        self.ru_oublock = int(ru_oublock)
+        self.ru_msgsnd = int(ru_msgsnd)
+        self.ru_msgrcv = int(ru_msgrcv)
+        self.ru_nsignals = int(ru_nsignals)
+        self.ru_nvcsw = int(ru_nvcsw)
+        self.ru_nivcsw = int(ru_nivcsw)
+
+@dataclass(init=False)
+class Path:
+    dirfd_minus_at_fdcwd: int
+    path: bytes
+    device_major: int
+    device_minor: int
+    inode: int
+    mtime: statx_timestamp
+    ctime: statx_timestamp
+    stat_valid: bool
+    dirfd_valid: bool
```

Comment:
Can we manually add the following `@property`:

```suggestion
    dirfd_valid: bool
    
    @property
    def dirfd() -> int:
        return self.dirfd_minus_at_fdcwd + AT_FDCWD
```

and elsewhere,

```python
# https://github.com/torvalds/linux/blob/73e931504f8e0d42978bfcda37b323dbbd1afc08/include/uapi/linux/fcntl.h#L98
AT_FDCWD = -100
```

Reply:
Currently, all the code for `Path` is generically autogenerated all the way from `prob_ops.h` to `ops.py`, adding the property would require some manual intervention somewhere, but it could be done (it would also require me to improve and generalize my python codegen code, but that's a me problem).

Reply:
added (along with generalized framework for adding custom properties) in https://github.com/charmoniumQ/PROBE/pull/20/commits/e1414d9fad0857e42556982558549d0609135dfd